src/google/protobuf/compiler/csharp/csharp_source_generator_base.cc

```diff
@@ -57,7 +59,12 @@ void SourceGeneratorBase::WriteGeneratedCodeAttributes(io::Printer* printer) {
 }
 
 std::string SourceGeneratorBase::class_access_level() {
```

Comment:
With this change, we can remove the IsDescriptorProto call here, and just make sure we call protoc appropriately in the generation code. (We still need the IsDescriptorProto function for other things though.)